op-service/signer/xlayer_auth.go

```diff
@@ -0,0 +1,217 @@
+package signer
+
+import (
+	"bytes"
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"net/http"
+	"sort"
+	"strings"
+)
+
+const (
+	headerSignKey   = "sign"
+	headerAccessKey = "accessKey"
+	algoSha256      = "sha256"
+)
+
+// addAuth adds authentication headers to the request
+// Uses the same authentication mechanism as xlayer-node: accessKey + SHA256 signature + AES encryption
+// If SecretKey or AccessKey is empty, no authentication is performed
+func (c *XLayerRemoteClient) addAuth(req *http.Request) error {
+	if c.config.SecretKey == "" || c.config.AccessKey == "" {
+		return nil
+	}
+
+	// Set accessKey header (use array syntax for consistency with xlayer-node)
+	req.Header[headerAccessKey] = []string{c.config.AccessKey}
+
+	// Generate signature using SHA256 algorithm
+	signature, err := c.genAuth(req, algoSha256)
+	if err != nil {
+		return fmt.Errorf("failed to generate signature: %w", err)
+	}
+
+	// Set sign header (use array syntax for consistency with xlayer-node)
+	req.Header[headerSignKey] = []string{signature}
+
+	return nil
+}
+
+// genAuth generates authentication signature from HTTP request
+func (c *XLayerRemoteClient) genAuth(req *http.Request, algorithm string) (string, error) {
+	// 1. Process URL query parameters
+	params := req.URL.Query()
+	treeMap := make(map[string][]string)
+	for _, v := range params {
```

Comment:
Only use values and discards parameter names, resulting in a severe security vulnerability.